### PR TITLE
OpenRewrite ReplaceLambdaWithMethodReference

### PR DESCRIPTION
**Pull request by koppor**
Manually applies the rule https://docs.openrewrite.org/recipes/java/cleanup/replacelambdawithmethodreference.

Not automatic possible, because of

- https://github.com/openrewrite/rewrite-static-analysis/issues/94
- https://github.com/openrewrite/rewrite-static-analysis/issues/95
- https://github.com/openrewrite/rewrite-static-analysis/issues/86#issuecomment-1537855430

```[tasklist]
### Compulsory checks
- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.
```

### CODE REVIEW

File: src/main/java/org/jabref/gui/JabRefFrame.java [hunk 1]
```diff
@@ -658,7 +658,7 @@ public void init() {
 
         // Bind global state
         FilteredList<Tab> filteredTabs = new FilteredList<>(tabbedPane.getTabs());
-        filteredTabs.setPredicate(tab -> tab instanceof LibraryTab);
+        filteredTabs.setPredicate(LibraryTab.class::isInstance);
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/JabRefFrame.java:661]: Oh that's cool! didn't know :)

File: src/main/java/org/jabref/logic/citationstyle/JabRefItemDataProvider.java [hunk 2]
```diff
@@ -191,7 +191,7 @@ public String toJson() {
         return entries.stream()
                 .map(entry -> bibEntryToCSLItemData(entry, bibDatabaseContext, entryTypesManager))
                 .map(item -> item.toJson(stringJsonBuilderFactory.createJsonBuilder()))
-                .map(item -> (String) item)
+                .map(String.class::cast)
```
Comment (calixtus) [inline src/main/java/org/jabref/logic/citationstyle/JabRefItemDataProvider.java:194]: Maybe this should have a comment that CSLItemData internally always provides a String?
Or should we test it before downcasting?

File: src/test/java/org/jabref/logic/shared/DBMSConnectionTest.java [hunk 3]
```diff
@@ -16,6 +16,17 @@ public class DBMSConnectionTest {
...
+                                .setType(dbmsType)
+                                .setHost("XXXX")
+                                .setPort(33778)
+                                .setDatabase("XXXX")
+                                .setUser("XXXX")
+                                .setPassword("XXXX")
+                                .setUseSSL(false)
+                                .setServerTimezone("XXXX")
+                                .createDBMSConnectionProperties())
+                        .getConnection());
```
Comment (calixtus) [inline src/test/java/org/jabref/logic/shared/DBMSConnectionTest.java:30]: Love it! ❤️

**Review by Siedlerchr — APPROVED** [review]

**Comment by Siedlerchr** [comment]
checkstyle

**Pull request merged**